src/Containers/ProductDetail.js

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useCallback } from 'react';
+import { useParams } from 'react-router-dom';
+import axios from 'axios';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectedProduct, removeSelectedProduct } from '../Redux/Actions/productActions';
+import '../Styles/ProductDetail.css';
+
+const ProductDetail = () => {
+  const product = useSelector((state) => state.product);
+  const {
+    image, title, price, category, description, rating,
+  } = product;
+  const { productid } = useParams();
+  const dispatch = useDispatch();
+
+  const fetchProductDetail = useCallback(async () => {
+    const response = await axios.get(`https://fakestoreapi.com/products/${productid}`).catch((err) => {
+      throw new Error('Err', err);
+    });
+    dispatch(selectedProduct(response.data));
+  }, [productid, dispatch]);
+
+  useEffect(() => {
+    if (productid && productid !== '') {
+      fetchProductDetail();
+    }
+
+    return () => {
+      dispatch(removeSelectedProduct());
+    };
+  }, [productid, fetchProductDetail, dispatch]); // Include dispatch in the dependency array
+
+  return (
+    <div className="DetailsPage">
+      {Object.keys(product).length === 0 ? (
+        <div><h3>...loading</h3></div>
+      ) : (
+        <>
+          <div className="Home-header">
+            <div className="image">
+              <img className="image" src={image} alt={title} />
+            </div>
+            <h2>
+              {title}
+            </h2>
+          </div>
+          <div className="productDetail">
+            <div className="productDescription">
+              <div className="productDescription">
+                Category:
+                {category}
+              </div>
+              <div className="productDescription">
+                Product Description:
+                {description}
+              </div>
+              <div className="productDescription">
+                Price:
+                $
+                {price}
+              </div>
+              <div className="productDescription">
+                Count:
+                {rating.count}
+              </div>
+              <div className="productDescription">
+                Rating:
+                {rating.rate}
+              </div>
+            </div>
+          </div>
+        </>
+      )}
+    </div>
```

Comment:
- [ ] Let's stick to the design as much as required. The page composition and placement of elements should follow the design. Kindly get rid of the double borders around the attributes. The attributes should stretch to the width ends of the screen.

- [ ] Kindly design the attributes so that the key value is aligned to the left while the value is aligned to the right end of the width as we have in the design.

- [ ] Kindly ensure you implement the stripe color pattern as we have in the design.

  Your Design | Behance Design
  :---:|:---:
  ![image](https://github.com/Dorcas126/capstone-project-metrics-webapp/assets/36542986/343c6a29-887e-4c49-95b3-a7faad450d7f) | ![details-2](https://github.com/Dorcas126/capstone-project-metrics-webapp/assets/36542986/ae879a00-ed3b-4c44-9c82-ae722246cdf2)

Reply:
Can be improved!